types/easycaptchajs/index.d.ts

```diff
@@ -0,0 +1,76 @@
+// Type definitions for easycaptchajs 1.1
+// Project: https://github.com/HichemTab-tech/EasyCaptchaJS#readme
+// Definitions by: HichemTab <https://github.com/HichemTab-tech>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace EasyCaptchaJs;
+
+export interface EasyCaptchaSettings {
+    ReCAPTCHA_API_KEY_CLIENT?: string;
+    ReCaptchaSubmit?: {
+        success?: () => void;
+        expired?: () => void;
+        failed?: () => void;
+    };
+    autoVerification?: {
+        okBtn: string | null;
+        requiredMsg: string;
+    };
+    apiScriptLoading?: {
+        loadingMsg: string;
+        error: () => void;
+        errorMsg: string;
+    };
+    theme?: 'light' | 'dark';
+    failure?: (error: string) => void;
+}
+
+export type EasyCaptchaMethod = 'getTarget' | 'verify' | 'response' | 'reset' | 'destroy';
+
+export type EasyCaptchaElementData = Record<string, any>;
+
+export interface EasyCaptchaResultBaseTarget {
+    parentElement?: JQuery,
+}
+
+export interface EasyCaptchaResultTarget extends EasyCaptchaResultBaseTarget {
+    data: EasyCaptchaElementData
+}
+
+export interface EasyCaptchaResultVerify extends EasyCaptchaResultBaseTarget {
+    verified: boolean
+}
+
+export interface EasyCaptchaResultResponse extends EasyCaptchaResultBaseTarget {
+    token: string
+}
+
+export type EasyCaptchaResult = EasyCaptchaResultTarget | EasyCaptchaResultVerify | EasyCaptchaResultResponse;
+
+export type EasyCaptchaResults = EasyCaptchaResult[] | EasyCaptchaResult | null;
+
+export interface EasyCaptchaInstance {
+    (options?: EasyCaptchaSettings, ...args: string[]): EasyCaptchaResults;
+    (method: EasyCaptchaMethod, results: any[], data: any, args: string[]): EasyCaptchaResults;
```

Comment:
isn't that 'data' to an an object with at least 'parentId'/'widgetId' properties? (I only point here, it should be at least `Record<string, unknown>` or similar, not just any, as it cannot be a scalar type, like boolean, number, etc

Reply:
Thanks for pointing that to me, i fixed it, now data have a defined type (`EasyCaptchaResult`) and not an `any`.